src/ui/window.cpp

```diff
@@ -723,6 +723,26 @@ void Window::onBuildTitleLabel()
   }
 }
 
+void trimTitleLabelBounds(Window* window)
+{
+  Widget* titleLabel = nullptr;
+  int mostLeftMiniButtonX = window->bounds().x2();
+  for (auto child : window->children()) {
+    if (child->type() == WidgetType::kWindowTitleLabelWidget)
+      titleLabel = child;
+    else if (child->isDecorative() && child->type() != WidgetType::kWindowTitleLabelWidget &&
+             child->bounds().x < mostLeftMiniButtonX)
+      mostLeftMiniButtonX = child->bounds().x;
+  }
+  if (!titleLabel)
+    return;
+  gfx::Rect newBounds(titleLabel->bounds());
+  newBounds.w = (titleLabel->bounds().x2() > mostLeftMiniButtonX ?
+                   mostLeftMiniButtonX - titleLabel->bounds().x :
+                   titleLabel->bounds().w);
+  titleLabel->setBounds(newBounds);
```

Comment:
This is just a matter of personal taste, not an actual thing to fix. In this specific case I believe that a regular `if` clause would improve readability:
```suggestion
  if (titleLabel->bounds().x2() > mostLeftMiniButtonX) {
    gfx::Rect newBounds(titleLabel->bounds());
    newBounds.w = mostLeftMiniButtonX - titleLabel->bounds().x;
    titleLabel->setBounds(newBounds);
  }
```